src/dmd/dclass.d

```diff
@@ -233,6 +233,9 @@ extern (C++) class ClassDeclaration : AggregateDeclaration
     /// specifies whether this is a D, C++, Objective-C or anonymous class/interface
     ClassKind classKind;
 
+    /// if this is a C++ class, this is the slot reserved for the virtual destructor
+    int cppDtorVtblIndex = -1;
```

Comment:
I'm not sure about the approach in this PR, as I'm not all that familiar with the way vtables are built in D.  However, we nee to maintain both a C++ and D interface to DMD (unless I'm mistaken both LDC and GDC use the C++ interface).  If you're adding a field to `ClassDelcaration`, I think you'll also need to update the same class in `aggregate.h` to keep the C++ interface compatible.  cc @ibuclaw and @kinke.

Reply:
updated it.